### PR TITLE
lint: add boolFuncPrefix checker

### LINKED ISSUE #230: lint/boolFuncPrefix: consider to name bool functions with `Is` prefix
name: `boolFuncPrefix `
before:
```go
func Enabled() bool
```
after:
```go
func IsEnabled() bool
```

### PR DESCRIPTION
**Pull request by cristaloleg**
Fixes #230

### CODE REVIEW

**Comment by Quasilyte** [comment]
What about `boolFuncPrefix` (`boolFuncPrefixChecker`)?
Flag makes it somewhat confusing because of `flag` package.

**Comment by cristaloleg** [comment]
Agreed. I'm not so good at naming things sometimes :(

**Comment by Quasilyte** [comment]
I also suggesting trying this out on GOROOT/src and catching corner cases that may don't need suggested prefix.

It will have many false positives for functions that are not predicates, but return bool to specify whether operation failed or not. Like `doRequest` or something like that. `isDoRequest` is not the right thing there (or is it?).

File: lint/flagFuncNaming_cheker.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+func (c *flagFuncNamingChecker) warn(fn *ast.FuncDecl) {
+	c.ctx.Warn(fn, "consider to add Is/Has/Contains prefix to function name")
+}
+
+func (c *flagFuncNamingChecker) isBoolType(expr ast.Expr) bool {
+	switch typ := c.ctx.TypesInfo.TypeOf(expr).(type) {
+	case *types.Basic:
+		return typ.Kind() == types.Bool
+	default:
+		return false
```
Comment (Quasilyte) [inline lint/flagFuncNaming_cheker.go]: (optional.)
Can do an `if` for a single case:
```go
if typ, ok := c.ctx.TypesInfo.TypeOf(expr).(*types.Basic); ok {
	return typ.Kind() == types.Bool
}
return false
```

Comment (Quasilyte) [reply lint/flagFuncNaming_cheker.go]: Even simpler:
```go
typ, ok := c.ctx.TypesInfo.TypeOf(expr).(*types.Basic)
return ok && return typ.Kind() == types.Bool
```

File: lint/boolFuncPrefix_cheker.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import (
+	"go/ast"
+	"go/types"
+	"strings"
+)
+
+func init() {
+	addChecker(boolFuncPrefixChecker{}, &ruleInfo{
+		Experimental: true,
+	})
```
Comment (Quasilyte) [inline lint/boolFuncPrefix_cheker.go]: `addChecker(boolFuncPrefixChecker{}, attrExperimental, attrVeryOpinionated)`

**Comment by Quasilyte** [comment]
Thanks for doing some investigation on other projects (I see new prefixes there).
LGTM when build is fixed.

**Pull request merged**